### DISCUSSION TITLE
Confused about which package to use.

### DISCUSSION THREAD

**Discussion by N6REJ** [ImageMagick/ImageMagick#5330] [Consulting]
Hello, let me start by admitting I'm pretty new to this whole dev integration gig.
I forked a [program](https://github.com/neard) due to its developer abandoning it and I'm having some confusion over exactly which upgrades to follow.
When I  use [windows.php.net](https://windows.php.net/downloads/pecl/deps/ImageMagick-7.1.0-18-vc15-x64.zip) the build.xml is happy.
but when I look at the [original repo ](https://github.com/neard/modules-untouched/releases/tag/php-r18) it shows "imagick 3.4.4" which doesn't include a bin at all.
we include imagemagick as a dependency for php

I've included a zip of the build.xml incase that helps clarify things.
The project is [here](https://github.com/Bearsampp) and is open to all

I'm sorry I sound so dumb, I'm an old man trying to understand.

[build.zip](https://github.com/ImageMagick/ImageMagick/files/9120421/build.zip)
.

**Comment by N6REJ** [comment]
I think i figured it out but could use some clarification....
so, apparently both 7.x and 3.4.x are required in different sections of the build folders.
![image](https://user-images.githubusercontent.com/1850089/179227182-24c6489f-91c5-4096-992b-18de489b8b99.png)

I'm a bit confused still over which of the [numerous dl's](https://imagemagick.org/script/download.php) to choose for the binary(?) part but chose https://imagemagick.org/archive/binaries/ImageMagick-7.1.0-43-Q16-x64-dll.exe and then https://windows.php.net/downloads/pecl/releases/imagick/3.7.0/php_imagick-3.7.0-8.1-ts-vs16-x64.zip
Doing this fixed the build.xml complaining.

So, if I'm understanding properly, I always need to update both together?

**Comment by snibgo** [comment]
> so, apparently both 7.x and 3.4.x are required ...

I suppose those versions are for two different products. "7.x" is likely to be ImageMagick, and "3.4.x" is probably IMagick.

IMagick provides the interface between PHP and ImageMagick. IMagick needs ImageMagick. ImageMagick does not need IMagick.

Sorry, I know nothing about PHP or IMagick.
